Makefile

```diff
@@ -1,7 +1,28 @@
-.PHONY: lint
+VENV=venv
+PIP_LOG=pip-log.txt
+PRECOMMIT_SRC=.githooks/pre-commit
+PRECOMMIT_DST=.git/hooks/pre-commit
+
+export VIRTUAL_ENV := $(abspath ${VENV})
+export PATH := ${VIRTUAL_ENV}/bin:${PATH}
+
+${PRECOMMIT_DST}: ${PRECOMMIT_SRC}
+	cp $< $@
+
+${PIP_LOG}: requirements.txt requirements-dev.txt
```

Comment:
uh, this is a bit weird. Why do you have a pip-log target rather than a venv one?

Reply:
So that we only run the code below when requirements have changed. To be frank, the target could just as well be `.timestamp-file-for-last-pip-install` and we could do a `touch $@` at the end with the same effect, but it felt like this is somewhat cleaner.

Now, if I instead rebuild the whole virtualenv whenever a requirement changes, then of course this is moot, since then the timestamp on the `venv` dir itself will work. (Contrast to not rebuilding the venv, just installing dependencies, when the timestamp on the dir doesn't change at installs, only when it's first created)